### ISSUE TITLE
rubygems documentation for "rational versioning" should reference Semantic Versioning

### ISSUE THREAD

**Issue by ColinDKelley** [rubygems/rubygems#3601]
I would like to suggest an improvement to the documentation.

While discussing https://github.com/rubygems/rubygems/issues/3574, it was clear that where the rubygems documentation refers to [rational versioning](https://github.com/rubygems/rubygems/issues/3574), that is now widely known as [Semantic Versioning](https://semver.org). I would like to update the documentation to use that term and point to that reference. The examples seem fine to leave. The hyperlink [here](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/version.rb#L131) is broken, however, so I'd suggest removing it.

Ok to send a PR?

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by simi** [comment]
That's indeed good idea.

**Comment by deivid-rodriguez** [comment]
Agreed, but we should make it clear (if it's not clear already) that our current operator is close but doesn't fully match semver specification.

For what it's worth, a fixed link would be https://www.zenspider.com/ruby/2008/10/rubygems-how-to-preventing-catastrophe.html, but I think I prefer to point to semver directly too.

**Comment by ColinDKelley** [comment]
Great, I'll put together a PR.

> Agreed, but we should make it clear (if it's not clear already) that our current operator is close but doesn't fully match semver specification.

I'll check that. I believe it's clearly documented as the "pessimistic version" matching . But I would like to add something explaining how to use it for SemVer matching.

There's also documentation over in the [Ruby Guides](https://guides.rubygems.org/patterns/). That already points to SemVer. Any tips on how to make a PR for that content?

**Comment by deivid-rodriguez** [comment]
I think that content's source is https://github.com/rubygems/guides/.

**Comment by ColinDKelley** [comment]
Aha, that's simple! Thanks @deivid-rodriguez!
Would one combined PR make sense, or would is there a reason to keep them separate?

**Comment by deivid-rodriguez** [comment]
I think I'm missing something because I thought you wanted to propose changes on two separate repos, so I'm not sure how you do that with a single PR :sweat_smile:.

**Comment by ColinDKelley** [comment]
Doh! I must have misread those repo paths and somehow thought they were both in the `rubygems/rubygems` repo.
I'll go with 2 PRs. 😄
